sycl/tools/sycl-xocc/bin/sycl_xocc.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python3
+"""sycl_xocc
+This is an extra layer of abstraction on top of the shell invocing the
+SDx compiler. As SPIR/LLVM-IR is a second class citizen in SDx for the
+moment it has some little niggling details that need worked on and the idea
+is that this script will work around those with some aid from Clang/LLVM.
+
+One of the main examples is that SDx can only compile one kernel from LLVM-BC
+at a time and it requires the kernels name (also required for kernel specific
+optimizations). This poses a problem as there can be multiple kernels in a
+file. And when making a normal naked SDx -c command in the driver
+you won't have the neccesarry information as the command is generated before
+the file's have even started to be compiled (Perhaps there is that I am
+unaware of). So no kernel names and no idea how many SDx commands you'd need
+to generate per file (no idea how many kernels are in a file).
+
+This works around that by using an opt (kernelNameGen) pass that generates
+an intermediate file with the needed information that we eat up and can then
+loop over each kernel in a file. It's simple at the moment just kernel names,
+but could expand in the future to include optimization information for each
+kernel.
+"""
+
+from argparse import ArgumentParser
+import json
+from os import environ
+from pathlib import Path
+import shutil
+import subprocess
+import tempfile
+
+
+class TmpDirManager:
+    def __init__(self, tmpdir: Path, prefix: str, autodelete: bool):
+        self.prefix = prefix
+        self.tmpdir = tmpdir
+        self.autodelete = autodelete
+
+    def __enter__(self) -> Path:
+        self.dir = Path(tempfile.mkdtemp(
+            dir=self.tmpdir,
+            prefix=self.prefix
+            ))
+        return self.dir
+
+    def __exit__(self, type, value, traceback):
+        if (self.autodelete):
+            shutil.rmtree(self.dir)
+
+
+class DataflowLawyerManager:
+    def __init__(self, vitis_clang_bin_dir: Path):
+        self.vitis_clang_bin_dir = vitis_clang_bin_dir
+        self.mounted = False
+
+    def __enter__(self):
+        self.dflawyer = self.vitis_clang_bin_dir / "xilinx-dataflow-lawyer"
+        echo = Path(shutil.which("echo"))
+        if self.dflawyer.is_file():
+            print(f"Binding {self.dflawyer} to {echo}")
+            subprocess.run([
+                "sudo", "mount", "--bind", echo, self.dflawyer
+            ])
+            self.mounted = True
+
+    def __exit__(self, _, __, ___):
```

Comment:
Should you type the `_`? :-)
Otherwise 
```suggestion
    def __exit__(self, *unused_args):
```
?

Reply:
Thanks. 
Will go with 

```
def __exit__(self, *_):
```